### ISSUE TITLE
node: assertion failure when calling parquet_metadata

### ISSUE THREAD

**Issue by zenazn** [duckdb/duckdb#6044]
### What happens?

When running the following SQL query from the nodejs bindings (and not e.g., the duckdb shell):

``` sql
select 'nothing' from parquet_metadata('/tmp/float.parquet')
```

I get the following assertion failure (and SIGABORT):
```
Assertion failed: (width >= Decimal::MAX_WIDTH_INT16 && width <= Decimal::MAX_WIDTH_INT32), function DECIMAL, file value.cpp, line 379.
```

We believe this is a regression from 0.6.1

### To Reproduce

First, make `/tmp/float.parquet` in the duckdb shell:

``` sql
copy (select 0.9) to '/tmp/float.parquet' (format 'parquet', codec 'zstd');
```

Then add a test that calls the above query. For instance, I added the following to `parquet.test.ts`:
``` typescript
    it('should be able to read parquet metadata', function(done) {
        db.all("select 'nothing' from parquet_metadata('/tmp/float.parquet')", done);
    });
```

Finally, run that test. For me, this command works: `npm test -- -g metadata`

### OS:

apple silicon OS X, but this originally happened on linux x64

### DuckDB Version:

0.6.2-dev* (I tried 45fed7847f427decf24539f487c70838009444e6)

### DuckDB Client:

Node

### Full Name:

Carl Jackson

### Affiliation:

Watershed

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by drewmcdonald** [comment]
Chiming in with a little more information from continuing to scope this with @zenazn. 

This does not parquet_metadata specific. Extending his original example above, this test will also fail:

```js
    it('should be able to query a parquet file', function(done) {
        db.all("select * from '/tmp/float.parquet', done);
    });
```

**Comment by jincongho** [comment]
### RCA

Parquet exported by the `COPY` has this schema:
```
message duckdb_schema {
  optional int32 0.9 (DECIMAL(2,1));
}
```
But failed the assertion in `value.cpp`:
```
Value Value::DECIMAL(int32_t value, uint8_t width, uint8_t scale) { // value=9, width=2, scale=1
	D_ASSERT(width >= Decimal::MAX_WIDTH_INT16 && width <= Decimal::MAX_WIDTH_INT32);
	Value result(LogicalType::DECIMAL(width, scale));
	result.value_.integer = value;
	result.is_null = false;
	return result;
}
```

### Solution

There's no `int16` primitive type in parquet, so I think generating `int32` with `decimal(2,1)` is correct. We just need to accommodate this in reading `ParquetStatisticsUtils::ConvertValue`:
```
case LogicalTypeId::DECIMAL: {
	auto width = DecimalType::GetWidth(type);
	auto scale = DecimalType::GetScale(type);
	switch (schema_ele.type) {
	case Type::INT32: {
		if (stats.size() != sizeof(int32_t)) {
			throw InternalException("Incorrect stats size for type %s", type.ToString());
		}
		return Value::DECIMAL(Load<int32_t>((data_ptr_t)stats.c_str()), width, scale);
	}
```

@Mytherin any thoughts?

**Comment by Mytherin** [comment]
Thanks for the report - I have pushed a fix in #6219. Indeed we update `Value::DECIMAL` to generate the correct value based on the width.
